osdep/terminal-unix.c

```diff
@@ -420,7 +420,7 @@ static void *terminal_thread(void *ptr)
             break;
         if (fds[1].revents) {
```

Comment:
IMO the proper fix would be to fix this check. It should be `if (fds[1].revents & POLLIN)` and there should be another check `if (fds[1].revents & POLLHUP) break;` after this block.